File: algorithms/1_easy/V/00941_valid_mountain_array/solution.py
class Solution:
    def validMountainArray(self, arr: List[int]) -> bool:
        n = len(arr)

        if n < 3 or arr[1] <= arr[0] or arr[n - 1] >= arr[n - 2]:
            return False

        index = 1

        while index < n and arr[index] > arr[index - 1]:
            index += 1

        if index == 1 or index == n or arr[index] == arr[index - 1]:
            return False

        while index < n and arr[index] < arr[index - 1]:
            index += 1

        return index == n

File: algorithms/1_easy/V/00941_valid_mountain_array/solution.cpp
class Solution {
public:
    bool validMountainArray(vector<int>& arr) {
        const size_t len{arr.size()};

        if (len < 3 || arr[1] <= arr[0] || arr[len - 1] >= arr[len - 2]) {
            return false;
        }

        size_t index{1};

        while (index < len && arr[index] > arr[index - 1]) {
            index += 1;
        }

        if (index == 1 || index == len || arr[index] == arr[index - 1]) {
            return false;
        }

        while (index < len && arr[index] < arr[index - 1]) {
            index += 1;
        }

        return index == len;
    }
};
